### ISSUE TITLE
Invalid code generation

### ISSUE THREAD

**Issue by cespare** [lazd/DOMly#27]
I've got some HTML that domly turns into JS which crashes. Here's a minimal repro (note the leading spaces in front of the `<script>` tag):

``` html
  <script src="a.js"></script>
```

I'm compiling with `domly.precompile(contents, {stripWhitespace: true})`.

I get this:

``` javascript
(function() {
  var frag;
function anonymous(data_0) {
    var frag = document.createDocumentFragment();
    var data = data_0;
    var el0 = document.createTextNode(" ");
    frag.appendChild(el0);
    frag.appendChild(el1);
    var el2 = document.createTextNode(" ");
    frag.appendChild(el2);
    return frag;
}
  return function template() {
    if (!frag) {
      frag = anonymous();
    }
    return frag.cloneNode(true);
  };
}())
```

See that `el1` is referenced but never created.

**Comment by lazd** [comment]
Looks Cheerio gives `<script>` tags a `type = 'script'`, not `type = 'tag'`, so the compiler completely missed them. The same goes for `<style>` tags. I've fixed both and added test cases.

**Comment by cespare** [comment]
Thanks for the quick fix!

**Comment by lazd** [comment]
Published as [`domly@0.0.8`](https://github.com/lazd/DOMly/releases/tag/v0.0.8).

**Comment by cespare** [comment]
Awesome!
